resources/voting-stats.js

```diff
@@ -0,0 +1,114 @@
+let voteChart = null;
+let data = [];
+let min_possible_vote = 1;
+let max_possible_vote = 50;
+
+function set_vote_graph_default_data(default_data, min_allowed_vote, max_allowed_vote) {
+    data = default_data;
+    min_possible_vote = min_allowed_vote;
+    max_possible_vote = max_allowed_vote;
+}
+
+function reload_vote_graph() {
+    if (voteChart !== null) voteChart.destroy();
+
+    if (data.length === 0) {
+        $('#id_no_votes_footer').prop('hidden', false);
+        $('#id_vote_chart').prop('hidden', true);
+        $('#id_has_votes_footer').prop('hidden', true);
+        $('#id_no_votes_error').prop('hidden', false);
+    } else {
+        $('#id_no_votes_footer').prop('hidden', true);
+        $('#id_vote_chart').prop('hidden', false);
+        $('#id_has_votes_footer').prop('hidden', false);
+        $('#id_no_votes_error').prop('hidden', true);
+
+        data.sort(function(a, b){return a - b});
+        // Give the graph some padding on both sides.
+        let min_points = Math.max(data[0] - 2, min_possible_vote);
+        let max_points = Math.min(data[data.length - 1] + 2, max_possible_vote);
+
+        let xlabels = [];
+        let voteFreq = [];
+        for (let i = min_points; i <= max_points; i++) {
+            xlabels.push(i);
+            voteFreq.push(0);
+        }
+
+        let max_number_of_votes = 0;
+        let total_votes = 0;
+        let mean = 0;
+
+        for (let i = 0; i < data.length; i++) {
+            // Assume the data is valid.
+            voteFreq[data[i] - min_points]++;
+            max_number_of_votes = Math.max(max_number_of_votes, voteFreq[data[i] - min_points]);
+            mean += data[i];
```

Comment:
I'm not in love with using a `mean` variable that _actually_ means `sum`. Let's split this up.

Reply:
Done.